### ISSUE TITLE
[REVIEW]: lifelines: survival analysis in Python

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1317]
**Submitting author:** @CamDavidsonPilon (<a href="http://orcid.org/0000-0003-1794-9143">Cameron Davidson-Pilon</a>)
**Repository:** <a href="https://github.com/camdavidsonpilon/lifelines" target ="_blank">https://github.com/camdavidsonpilon/lifelines</a>
**Version:** v0.22.2
**Editor:** @trallard
**Reviewer:** @becarioprecario, @sunhwan
**Archive:** <a href="https://doi.org/10.5281/zenodo.805993" target="_blank">10.5281/zenodo.805993</a>

## Status

[![status](http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76/status.svg)](http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76"><img src="http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76/status.svg)](http://joss.theoj.org/papers/cd6f19605906416d5196f9cdd3841a76)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/camdavidsonpilon/lifelines" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@becarioprecario & @sunhwan, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @trallard know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/camdavidsonpilon/lifelines">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.22.2
- [x] **Authorship:** Has the submitting author (@CamDavidsonPilon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/camdavidsonpilon/lifelines">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.22.2
- [x] **Authorship:** Has the submitting author (@CamDavidsonPilon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @becarioprecario, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01317/joss.01317/10.21105.joss.01317.pdf)

**Comment by trallard** [comment]
👋 @becarioprecario and @sunhwan this is the issue for the review. Each of you has a checklist of the items you have to look at while conducting your review. 

Please use this issue to comment, make suggestions and discuss anything related to the submission. If you need to request changes or go in detail on something it is advisable to make an issue on the software repo [https://github.com/camdavidsonpilon/lifelines](https://github.com/camdavidsonpilon/lifelines) and reference this issue to keep track of the process. 

If any of you or @CamDavidsonPilon have any questions at any point feel free to ping me.

Happy reviewing 🔎💻

**Comment by trallard** [comment]
Hi @becarioprecario and @sunhwan I have not seen much activity on this repo for a bit so I was wondering if there is anything you need help with?

**Comment by sunhwan** [comment]
Thanks for reminding me. I'll work on this soon.

**Comment by becarioprecario** [comment]
Thanks for reminding me. I’ll try to look into this asap.

Virgilio

**Comment by sunhwan** [comment]
@trallard I went through the checklist and I was satisfied with most of the item in the list. I found a few problems running some script/file. I left a comment in the repository. https://github.com/CamDavidsonPilon/lifelines/issues/703 

This package looks great and I'm okay to accept after the author address the minor issue. Are reviewers expected to write a separate review?

**Comment by trallard** [comment]
Fantastic @sunhwan thanks a lot for your review.

@becarioprecario  is there anything we can help with to move this review forward ?

**Comment by kyleniemeyer** [comment]
Hi @becarioprecario, just wanted to check on your review.

**Comment by trallard** [comment]
ping! @becarioprecario  do you have any updates on the review?

**Comment by becarioprecario** [comment]
Hi,

I have been a bit stuck with other stuff. Please, may I ask you to send me the guidelines again on how to make the review.

Thanks!

Virgilio

**Comment by trallard** [comment]
Hi @becarioprecario  the checklist for the review is at the top of this issue 👆🏼

For reference the guidelines for reviewers are located here: [https://joss.theoj.org/about#reviewer_guidelines](https://joss.theoj.org/about#reviewer_guidelines)

**Comment by becarioprecario** [comment]
Hi,

I am done with my review now. The package and its documentation are excellent!! I had some troubles with the installation, but due to my local configuration. Installing the package with pip run smoothly.

Best,

Virgilio

**Comment by trallard** [comment]
Thanks for your time and reviews @becarioprecario  and @sunhwan 🙏🏼

@CameronDevine I can confirm your paper has been accepted now! 🎉  Let's move this to publication, please complete the following actions and let me know once this is done

- Final check on article: DOIS are present, authors are correctly listed
- Create a  new tag for the software and add let me know what final version is
- Deposit the software in Zenodo and share the DOI here

**Comment by trallard** [comment]
@whedon  commands

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01317/joss.01317/10.21105.joss.01317.pdf)

**Comment by trallard** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5281/zenodo.275519 is OK
- 10.5281/zenodo.1472929 is OK
- 10.1111/j.1751-5823.2009.00095_1.x is OK
- 10.1111/ajps.12176 is OK
- 10.1002/sim.2864 is OK
- 10.1007/s10654-016-0149-3 is OK
- 10.1111/j.1751-5823.2009.00095_1.x is OK
- 10.1007/s40471-016-0089-1 is OK
- 10.18637/jss.v070.i08 is OK

MISSING DOIs

- https://doi.org/10.1002/(sici)1097-0258(19960229)15:4<361::aid-sim168>3.0.co;2-4 may be missing for title: Multivariable prognostic models: issues in developing models, evaluating assumptions and adequacy, and measuring and reducing errors.

INVALID DOIs

- None
```

**Comment by CameronDevine** [comment]
@CamDavidsonPilon, I just wanted to make sure you saw this since I was mentioned by mistake.

**Comment by trallard** [comment]
HI @CamDavidsonPilon I was wondering if you'd make any progress with the last few tasks for acceptance?

**Comment by CamDavidsonPilon** [comment]
hi @trallard, I have addressed the missing DOI. Anything else? Is there a list I missed?

**Comment by trallard** [comment]
Can you please paste the DOI here so I can set it as the archive? As well as the latest version of the package and I can finalise the acceptance

**Comment by CamDavidsonPilon** [comment]
Missing DOI entry: https://github.com/CamDavidsonPilon/lifelines/blob/master/paper/paper.bib#L144-L153

Latest version is [0.21.1](https://github.com/CamDavidsonPilon/lifelines/blob/155e9bb9e913e19858a91885a36ac844931c1eee/lifelines/version.py#L4)

**Comment by trallard** [comment]
@whedon set version as 0.21.1

**Comment by trallard** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01317/joss.01317/10.21105.joss.01317.pdf)

**Comment by trallard** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by trallard** [comment]
@whedon set 0.21.1 as version

**Comment by whedon** [comment]
OK. 0.21.1 is the version.

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.5281/zenodo.275519 is OK
- 10.5281/zenodo.1472929 is OK
- 10.1111/j.1751-5823.2009.00095_1.x is OK
- 10.1111/ajps.12176 is OK
- 10.1002/sim.2864 is OK
- 10.1007/s10654-016-0149-3 is OK
- 10.1111/j.1751-5823.2009.00095_1.x is OK
- 10.1007/s40471-016-0089-1 is OK
- https://doi.org/10.1002/(SICI)1097-0258(19960229)15:4<361::AID-SIM168>3.0.CO;2-4 is OK
- 10.18637/jss.v070.i08 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by trallard** [comment]
@whedon set 10.5281/zenodo.805993 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.805993" target="_blank">10.5281/zenodo.805993</a> is the archive.

**Comment by trallard** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/847

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/847, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by trallard** [comment]
@CamDavidsonPilon Lifelines has been accepted to JOSS 🎉🎉

@becarioprecario and @sunhwan thanks for the time and effort put in this review. Your contribution to JOSS is deeply appreciated. 🙏🏼🙌🏼🙂

@openjournals/joss-eics this paper is ready for acceptance 👾

**Comment by trallard** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @trallard, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by danielskatz** [comment]
@trallard - you've jumped the gun slightly.  Let's say that you believe Lifelines is ready to be accepted to JOSS - now @openjournals/joss-eics need to do some final checks.

**Comment by trallard** [comment]
ooops wrong wording you are correct!!! sorry!

**Comment by danielskatz** [comment]
@CamDavidsonPilon - please accept the changes in https://github.com/CamDavidsonPilon/lifelines/pull/781, or disagree with the ones that are not adding needed spaces :)

**Comment by danielskatz** [comment]
Sorry for the delay in seeing this - I was expecting you to say you had merged the PR, which would have led to a GH notification.  @labarba - this was in the midst of acceptance, perhaps as the on-duty @openjournals/joss-eics you can continue the process

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/879

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/879, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@CamDavidsonPilon — the listing on the bottom of page 2 bleeds into the margin. Could you add some line breaks to make that render better?

**Comment by labarba** [comment]
Also, I had a look at the Zenodo archive here: http://doi.org/10.5281/zenodo.805993
It shows a long list of authors, possibly an automatic listing from the commits. 

We request that authors modify the Zenodo-archive metadata so that the title and author list match the JOSS paper. You may also want to add your ORCID to the metadata.

**Comment by CamDavidsonPilon** [comment]
Okay, both those should be done @labarba

**Comment by labarba** [comment]
I see your latest tagged release is v0.22.2, but here we have 0.21.1 — could you double check which version we should be associating the JOSS paper with?

**Comment by labarba** [comment]
Also, the Zenodo archive shows the repo name as title. Could you edit the metadata to match the JOSS paper title?

**Comment by CamDavidsonPilon** [comment]
> I see your latest tagged release is v0.22.2, but here we have 0.21.1 — could you double check which version we should be associating the JOSS paper with?

If I change the JOSS paper to be associated to 0.22.2, does it need to be re-reviewed, etc.? If not, v0.22.2 is correct.

**Comment by labarba** [comment]
@whedon set v0.22.2 as archive

**Comment by whedon** [comment]
v0.22.2 doesn't look like an archive DOI.

**Comment by labarba** [comment]
@whedon set v0.22.2 as version

**Comment by whedon** [comment]
OK. v0.22.2 is the version.

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/882

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/882, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@CamDavidsonPilon — just one more tweak on the comment line that bleeds into the margin, please!

**Comment by labarba** [comment]
You can run `@whedon generate pdf` right here to see the proof.

**Comment by CamDavidsonPilon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01317/joss.01317/10.21105.joss.01317.pdf)

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1158065976501526529) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/883
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01317](https://doi.org/10.21105/joss.01317)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @CamDavidsonPilon, your JOSS paper is published! 🚀 

Huge thanks to our editor: @trallard, and the reviewers: @becarioprecario, @sunhwan — we greatly appreciate your contribution to JOSS 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01317/status.svg)](https://doi.org/10.21105/joss.01317)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01317">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01317/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01317/status.svg
   :target: https://doi.org/10.21105/joss.01317
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01317/status.svg)](https://doi.org/10.21105/joss.01317)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by CamDavidsonPilon** [comment]
Thanks @labarba, @trallard, @becarioprecario and @sunhwan!
